Systems/WorldGen/Standard/ChunkGen/1.GenTerra/GenTerra.cs

```diff
@@ -132,6 +135,18 @@ public void initWorldGen()
             paddedNoiseWidth = noiseWidth + 1;
             paddedNoiseHeight = noiseHeight + 1;
 
+            // Until v1.18.0, a few key calculations used paddedNoiseWidth when they should have used noiseWidth.
+            if (GameVersion.IsAtLeastVersion(api.WorldManager.SaveGame.CreatedGameVersion, "1.18.0-pre.1"))
+            {
+                previouslyPaddedNoiseWidth = noiseWidth;
+                previouslyWeirdMultiplier = 1.0f;
+            }
+            else
+            {
+                previouslyPaddedNoiseWidth = paddedNoiseWidth;
+                previouslyWeirdMultiplier = (float)lerpHor / chunksize + 1.0f;

```

Comment:
`chunkPixelSize` not accessible here, so factored out to turn `weirdOffset` into a multiplier instead.